src/adagrams.js

```diff
@@ -1,15 +1,121 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
 export const drawLetters = () => {
-  // Implement this method for wave 1
+
+  let drawedLetters = [];
+  const numAllowedinHand = 10;
+  let allLetters = [];
+
+  for (const [key, value] of Object.entries(LETTER_POOL)){
+    for (let i = 0; i < value; i++){
+      allLetters.push(key);
+    };
+  };
+
+  while (drawedLetters.length < numAllowedinHand){
+    let randomNumber = Math.floor(Math.random() * allLetters.length);
+    drawedLetters.push(allLetters[randomNumber]);
+    allLetters.splice(randomNumber,1);
```

Comment:
Nitpick: missing whitespace
```suggestion
    allLetters.splice(randomNumber, 1);
```

Reply:
What is the time complexity for the array `splice` method? If you have a list of elements and have to randomly remove one, then you need to shift all the other elements to fill the empty gap left by the element you just spliced. 

`splice` like python's `remove` has linear time complexity with respect to the length of the list. This won't be a problem for the size of data we're working with, but consider using a strategy of that would be constant time. 

An idea could be to swap the randomly drawn letter to the end of the list (constant time operation), then pop the random letter that is now at the end (constant time operation).